### PR TITLE
(feat) Add Print Functionality to ESM Common Library with React

### PR DESCRIPTION
**Pull request by nanfuka**
## Requirements

- [x] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes a [conventional commit](https://o3-dev.docs.openmrs.org/#/getting_started/contributing?id=your-pr-title-should-indicate-the-type-of-change-it-is) label. See existing PR titles for inspiration.
- [x] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide) and [design documentation](https://zeroheight.com/23a080e38/p/880723-introduction).
- [ ] My work includes tests or is validated by existing tests.

## Summary
This pull request introduces the PrintComponent component, enabling the generation of printable documents with customizable content

## Screenshots
Screen Shot 2023-06-14 at 8 48 10 AM

## Related Issue

https://issues.openmrs.org/browse/O3-2192

### CODE REVIEW

**Comment by gracepotma** [comment]
Tagged you @mogoodrich since I know PIH uses +++ printing functionality, so would be great just to check w/ you that this is heading in the direction your team will need. 

Huge thanks to you Deb!! <3

**Comment by vasharma05** [comment]
Hi @nanfuka !
I actually had a point to share on the print

Is it okay for table to be printed with pagination?
Shouldn't we expand and show all data at hand for the print feature?

**Comment by nanfuka** [comment]
That was an oversight. It seems that all the available data should be included in the printout. Thank you for bringing this to my attention, @vasharma05.

**Comment by mogoodrich** [comment]
Thanks @gracepotma !  Yes, we'd definitely like printing.  @nanfuka  is there a ticket or specification or design laying out the full requirements for this PR?

**Comment by nanfuka** [comment]
@mogoodrich this is the [epic,](https://issues.openmrs.org/browse/O3-2188) you will see the features that fall within that epic

File: packages/esm-patient-common-lib/src/print/print.component.tsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+export function PrintComponent({ subheader, body, patientDetails, requestedBy }) {
+  const componentRef = useRef(null);
+  const { t } = useTranslation();
+  const handlePrint = useReactToPrint({
+    content: () => componentRef.current,
+  });
+
+  return (
+    <div>
+      <Button kind="ghost" renderIcon={Printer} iconDescription="Add vitals" onClick={handlePrint}>
```
Comment (mogoodrich) [inline packages/esm-patient-common-lib/src/print/print.component.tsx:18]: The text here I assume should be something print related, not "Add Vitals"?

File: packages/esm-patient-common-lib/src/print/print.component.tsx [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        {t('print', 'Print')}
```
Comment (mogoodrich) [inline packages/esm-patient-common-lib/src/print/print.component.tsx:19]: Did we add the "print" message code to the English translations file?

File: packages/esm-patient-common-lib/src/print/print.component.tsx [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+      </Button>
+      <div className={`${styles.printpage} `}>
+        <div className={`${styles} `} ref={componentRef}>
+          <Header>
+            <img src={log} alt="logo" width={110} height={40} className={styles.img} />
+          </Header>
+
+          <div className={styles.patientDetails}>
+            <p>{patientDetails?.name}</p>
+            <p>{patientDetails?.age} years</p>
```
Comment (mogoodrich) [inline packages/esm-patient-common-lib/src/print/print.component.tsx:29]: We should localize "years"

File: packages/esm-patient-common-lib/src/print/print.component.tsx [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+  return (
+    <div>
+      <Button kind="ghost" renderIcon={Printer} iconDescription="Add vitals" onClick={handlePrint}>
+        {t('print', 'Print')}
+      </Button>
+      <div className={`${styles.printpage} `}>
+        <div className={`${styles} `} ref={componentRef}>
+          <Header>
+            <img src={log} alt="logo" width={110} height={40} className={styles.img} />
```
Comment (mogoodrich) [inline packages/esm-patient-common-lib/src/print/print.component.tsx:24]: It looks like this always prints the OpenMRS logo?  I assume we want to use whatever logo a particular implementation has configured?

File: packages/esm-patient-common-lib/src/print/print.component.tsx [hunk 5]
```diff
@@ -0,0 +1,48 @@
+import React, { useRef } from 'react';
+import { Printer } from '@carbon/react/icons';
+import { useReactToPrint } from 'react-to-print';
+import { useTranslation } from 'react-i18next';
+import styles from './print.scss';
+import { Header, Button } from '@carbon/react';
+import log from './log.png';
+
+export function PrintComponent({ subheader, body, patientDetails, requestedBy }) {
```
Comment (mogoodrich) [inline packages/esm-patient-common-lib/src/print/print.component.tsx:9]: Do we have an example usage of how these props should be populated?  For example, is "patientDetails" a standard data structure we define or us elsewhere?

**Review by mogoodrich — COMMENTED** [review]
I added few comments on this, but mainly defer to @denniskigen  and others as I'm not really familar with react-to-print or exactly how this feature is suppose to work.

File: packages/esm-patient-common-lib/src/print/log.png
Comment (denniskigen) [inline packages/esm-patient-common-lib/src/print/log.png:1]: Should be renamed to `logo.png`. Probably best to use an SVG version instead now I think about it.

**Comment by hadijahkyampeire** [comment]
@nanfuka I am gonna close this since you already did the print work.

**Pull request closed without merging**